src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2StreamTests.cs

```diff
@@ -1839,6 +1839,32 @@ public async Task ResponseTrailers_WithoutData_Sent()
             Assert.Equal("Custom Value", _decodedHeaders["CustomName"]);
         }
 
+        [Fact]
+        public async Task ResponseTrailers_WithExeption500_Cleared()
+        {
+            await InitializeConnectionAsync(context =>
+            {
+                context.Response.AppendTrailer("CustomName", "Custom Value");
```

Comment:
Do we have any tests where the App attempts to call AppendTrailer after CompleteAsync. That should throw an InvalidOperationException due to being read-only at that point, right?

Reply:
I would test both code paths. One where a trailer was added before CompleteAsync, and another where no trailers were written so the empty read-only trailer header collection is lazy-initialized.

Reply:
These tests check IsReadOnly on the trailers collection. I could also try adding in one of them.

Reply:
It's fine either way. I didn't see those tests.